diff --git a/firmware/python/epix_uhr_gtreadout_dev/__init__.py b/firmware/python/epix_uhr_gtreadout_dev/__init__.py
index 2a4b9b4..452e23e 100644
--- a/firmware/python/epix_uhr_gtreadout_dev/__init__.py
+++ b/firmware/python/epix_uhr_gtreadout_dev/__init__.py
@@ -1,3 +1,4 @@
+./firmware/python/epix_uhr_gtreadout_dev/__init__.py
 from epix_uhr_gtreadout_dev._App                           import *
 from epix_uhr_gtreadout_dev._Root                          import *
 from epix_uhr_gtreadout_dev._UhrAxiStreamFramer            import *
diff --git a/setup.py b/setup.py
new file mode 100644
index 0000000..f7934ac
--- /dev/null
+++ b/setup.py
@@ -0,0 +1,90 @@
+from setuptools import setup
+
+# use softlinks to make the various "board-support-package" submodules
+# look like subpackages.  Then __init__.py will modify
+# sys.path so that the correct "local" versions of surf etc. are
+# picked up.  A better approach would be using relative imports
+# in the submodules, but that's more work.  -cpo
+#usesoftlinkstomakethevarious"board-support-package"submodules
+#looklikesubpackages.Then__init__.pywillmodify
+#sys.pathsothatthecorrect"local"versionsofsurfetc.are
+#pickedup.Abetterapproachwouldbeusingrelativeimports
+#inthesubmodules,butthat'smorework.-cpo
+setup(
+    name = 'epix_uhr_dev',
+    description = 'Epix UHR development package',
+    packages = [
+'epix-uhr-dev',
+'epix-uhr-dev.l2si_core',
+'epix-uhr-dev.epix_hr_leap_common',
+'epix-uhr-dev.epix_hr_core',
+'epix-uhr-dev.ePixViewer',
+'epix-uhr-dev.ePixViewer.asics',
+'epix-uhr-dev.LclsTimingCore',
+'epix-uhr-dev.axipcie',
+'epix-uhr-dev.asic_reg_mapping',
+'epix-uhr-dev.asic_reg_mapping.EpixHr10kT',
+'epix-uhr-dev.asic_reg_mapping.SparkPixS',
+'epix-uhr-dev.asic_reg_mapping.EpixUhr100k',
+'epix-uhr-dev.asic_reg_mapping.SparkPixED',
+'epix-uhr-dev.asic_reg_mapping.Tixel',
+'epix-uhr-dev.asic_reg_mapping.Epix10ka',
+'epix-uhr-dev.asic_reg_mapping.EpixS',
+'epix-uhr-dev.asic_reg_mapping.EpixHrAdc',
+'epix-uhr-dev.asic_reg_mapping.SparkPixS_XPC',
+'epix-uhr-dev.asic_reg_mapping.SparkPixRt',
+'epix-uhr-dev.asic_reg_mapping.EpixUhr',
+'epix-uhr-dev.asic_reg_mapping.Cpix2',
+'epix-uhr-dev.asic_reg_mapping.Epix100a',
+'epix-uhr-dev.asic_reg_mapping.EpixHrMv2',
+'epix-uhr-dev.surf.ethernet',
+'epix-uhr-dev.surf.ethernet.mac',
+'epix-uhr-dev.surf.ethernet.gige',
+'epix-uhr-dev.surf.ethernet.udp',
+'epix-uhr-dev.surf.ethernet.ten_gig',
+'epix-uhr-dev.surf.ethernet.xaui',
+'epix-uhr-dev.surf.dsp',
+'epix-uhr-dev.surf.dsp.fixed',
+'epix-uhr-dev.surf.axi',
+'epix-uhr-dev.surf.misc',
+'epix-uhr-dev.surf.devices.ti',
+'epix-uhr-dev.surf.devices.amphenol',
+'epix-uhr-dev.surf.devices.maxim',
+'epix-uhr-dev.surf.devices.nxp',
+'epix-uhr-dev.surf.devices.silabs',
+'epix-uhr-dev.surf.devices.linear',
+'epix-uhr-dev.surf.devices.intel',
+'epix-uhr-dev.surf.devices.transceivers',
+'epix-uhr-dev.surf.devices.cypress',
+'epix-uhr-dev.surf.devices.microchip',
+'epix-uhr-dev.surf.devices.micron',
+'epix-uhr-dev.surf.devices.analog_devices',
+'epix-uhr-dev.surf.devices',
+'epix-uhr-dev.surf',
+'epix-uhr-dev.surf.protocols.batcher',
+'epix-uhr-dev.surf.protocols.pgp',
+'epix-uhr-dev.surf.protocols.jesd204b',
+'epix-uhr-dev.surf.protocols.rssi',
+'epix-uhr-dev.surf.protocols.ssp',
+'epix-uhr-dev.surf.protocols.ssi',
+'epix-uhr-dev.surf.protocols.i2c',
+'epix-uhr-dev.surf.protocols',
+'epix-uhr-dev.surf.protocols.htsp',
+'epix-uhr-dev.surf.protocols.sugoi',
+'epix-uhr-dev.surf.protocols.clink',
+'epix-uhr-dev.surf.protocols.coaxpress',
+'epix-uhr-dev.surf.xilinx',
+],
+package_dir = {
+'epix-uhr-dev'                     : 'firmware/python/epix_uhr_gtreadout_dev',
+'epix-uhr-dev.l2si_core'           : 'firmware/submodules/l2si-core/python/l2si_core',
+'epix-uhr-dev.epix_hr_leap_common' : 'firmware/submodules/epix-hr-leap-common/python/epix_hr_leap_common',
+'epix-uhr-dev.epix_hr_core'        : 'firmware/submodules/epix-hr-core/python/epix_hr_core',
+'epix-uhr-dev.ePixViewer'          : 'firmware/submodules/ePixViewer/python/ePixViewer',
+'epix-uhr-dev.ePixViewer.asics'    : 'firmware/submodules/ePixViewer/python/ePixViewer/asics',
+'epix-uhr-dev.LclsTimingCore'      : 'firmware/submodules/lcls-timing-core/python/LclsTimingCore',
+'epix-uhr-dev.axipcie'             : 'firmware/submodules/axi-pcie-core/python/axipcie',
+'epix-uhr-dev.asic_reg_mapping'    : 'firmware/submodules/AsicRegMapping/python/asic_reg_mapping',
+'epix-uhr-dev.surf'                : 'firmware/submodules/surf/python/surf',
+}
+)
